addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,366 @@
+                                || val.setScale(4, RoundingMode.HALF_UP).equals(maxInt16AsInt)) {
+                            continue;
+                        }
+                    }
+                    /* Check the capabilities of this service */
+                    if (serObj.serviceTreeMap.get(subKey).getValueParameter() != null) {
+                        /* Creating an new channel type with capabilities from service */
+                        @SuppressWarnings("unchecked")
+                        List<BigDecimal> subValParas = (List<BigDecimal>) serObj.serviceTreeMap.get(subKey)
+                                .getValueParameter();
+                        BigDecimal minVal = subValParas.get(0);
+                        BigDecimal maxVal = subValParas.get(1);
+
+                        StateDescription state = new StateDescription(minVal, maxVal, null, "%.2f", false, null);
+
+                        ChannelTypeUID channelTypeUID = new ChannelTypeUID(thing.getUID().getBindingId(),
+                                CHANNEL_DYNAMIC_NUMBER);
+
+                        ChannelType channelType = new ChannelType(channelTypeUID, false, "NumberType", subKey, subKey,
+                                null, null, state, null);
+
+                        factory.addChannelType(channelType);
+
+                        Channel newChannel = ChannelBuilder
+                                .create(new ChannelUID(getThing().getUID(), subNameAddon + subKey), "Number")
+                                .withType(channelTypeUID).withLabel(subKey).withKind(ChannelKind.STATE)
+                                .withProperties(properties).build();
+                        subChannels.add(newChannel);
+
+                    } else {
+                        Channel newChannel = ChannelBuilder
+                                .create(new ChannelUID(thing.getUID(), subNameAddon + subKey), "Number")
+                                .withType(new ChannelTypeUID(thing.getUID().getBindingId(), CHANNEL_FLOAT_VALUE))
+                                .withDescription(subKey).withLabel(subKey).withKind(ChannelKind.STATE)
+                                .withProperties(properties).build();
+                        subChannels.add(newChannel);
+                    }
+                    break;
+
+                case "refEnum":
+                    /* Check whether the sub service should be ignored */
+                    Boolean ignoreIt = false;
+                    for (KM200ThingTypes tType : KM200ThingTypes.values()) {
+                        if (tType.getThingTypeUID().equals(thing.getThingTypeUID())) {
+                            for (String ignore : tType.ignoreSubService()) {
+                                if (ignore.equals(subKey)) {
+                                    ignoreIt = true;
+                                }
+                            }
+
+                        }
+                    }
+                    if (ignoreIt) {
```

Comment:
lines 324 to 334 might be a separate method so this will be a one-liner here.

Reply:
I did some code changes here. It should be better now.